### PR TITLE
Category for guides on the API

### PR DESCRIPTION
**Pull request by dariomtz**
- The POST guide endpoint now has a category parameter.
- The GET method filters through category if a category parameter is provided.
- Implements the necessary changes on DatastoreGuide and Guide class.
- Documents the changes on the README.md file.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/GuidesServlet.java [hunk 1]
```diff
@@ -23,16 +23,25 @@
...
         response.setContentType("application/json");
-        response.getWriter().print(gson.toJson(helper.queryAll()));
+        String category = request.getParameter("category");
+
+        if(category == null){
+            response.getWriter().print(gson.toJson(helper.queryAll()));
+            return;
+        }
+        
+        category = Jsoup.clean(category, Whitelist.none());
```
Comment (scunning-sps) [inline src/main/java/com/google/sps/servlets/GuidesServlet.java:33]: What happens if this is an empty string? Does it filter the results, or return all guides?

File: src/main/java/com/google/sps/servlets/GuidesServlet.java [hunk 2]
```diff
@@ -23,16 +23,25 @@
...
+        response.getWriter().print(gson.toJson(helper.queryByCategory(category)));
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
         String title = Jsoup.clean(request.getParameter("title"), Whitelist.none());
         String description = Jsoup.clean(request.getParameter("description"), Whitelist.none());
         String content = Jsoup.clean(request.getParameter("content"), Whitelist.none());
+        String category = Jsoup.clean(request.getParameter("category"), Whitelist.none());
```
Comment (scunning-sps) [inline src/main/java/com/google/sps/servlets/GuidesServlet.java:42]: Similarly, do you want to store the category as an empty string or null?

**Review by scunning-sps — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/GuidesServlet.java (hunk 1 above)
Comment (dariomtz) [reply src/main/java/com/google/sps/servlets/GuidesServlet.java:33]: As it is, it returns the guides that have category set to an empty string. But what I am thinking about this is that category is a mandatory field.

File: src/main/java/com/google/sps/servlets/GuidesServlet.java (hunk 2 above)
Comment (dariomtz) [reply src/main/java/com/google/sps/servlets/GuidesServlet.java:42]: Well, I am thinking of every parameter to be mandatory, so I should be checking that none of them are null or empty strings and return bad request if any of them are.

**Pull request merged**